discussion_forum/tests.py

```diff
@@ -0,0 +1,592 @@
+
+from .models import DiscussionComment, DiscussionReply, DiscussionThread
+
+
+ins_cred = {"email": "instructor@bodhitree.com", "password": "instructor"}
+
+
+class DiscussionThreadViewSetTest(APITestCase):
+    """Test for DiscussionThreadViewSet."""
+
+    fixtures = [
+        "users.test.yaml",
+        "courses.test.yaml",
+        "coursehistories.test.yaml",
+        "discussionforum.tests.yaml",
+        "discussionthread.tests.yaml",
+    ]
+
+    def _list_discussion_threads_helper(self):
+        """helper function to test list discussion threads functionality."""
+
+        url = reverse(
+            "discussion_forum:discussionthread-list-discussion-threads", args=[1]
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        length = DiscussionThread.objects.all().count()
+        self.assertEqual(len(response.data), length)
+
+    def test_list_discussion_threads(self):
+        """
+        Test to check: list all discussion_threads.
+        """
+        self.client.login(**ins_cred)
+        self._list_discussion_threads_helper()
+        self.client.logout()
+
+    def _retrieve_discussion_thread_helper(self, discussion_thread_id):
+        """helper function to test retrive discussion thread functionality."""
+
+        url = reverse(
+            "discussion_forum:discussionthread-retrieve-discussion-thread",
+            kwargs={"pk": discussion_thread_id},
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["id"], discussion_thread_id)
+
+    def test_retrieve_discussion_thread(self):
+        """
+        Test to check: retrieve discussion_thread.
+        """
+        discussion_thread_id = 1
+        self.client.login(**ins_cred)
+        self._retrieve_discussion_thread_helper(discussion_thread_id)
+        self.client.logout()
+
+    def _create_discussion_thread_helper(self, title, status_code):
+        """helper function to test create discussion thread functionality.
+
+        Args:
+        title (str): title of the discussion thread
+        status_code (int): expected status code of the API call
+        """
+
+        data = {
+            "discussion_forum": 1,
+            "title": title,
+            "mark_as_important": True,
+            "author": 1,
+            "author_category": "I",
+            "description": "Description of discussion thread",
+            "pinned": True,
+            "anonymous_to_student": True,
+            "upvote": 0,
+            "downvote": 0,
+        }
+        url = reverse(
+            "discussion_forum:discussionthread-create-discussion-thread", args=[1]
+        )
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status_code)
+        if status_code == status.HTTP_201_CREATED:
+            return_data = response.data
+            for k in [
```

Comment:
use proper name for k like field. also use response_data instead of return_data

Reply:
done